sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -53,6 +56,35 @@ def _check_for_matching_filetype(pattern, filename):
     return file_dictionary
 
 
+def create_path_to_upload(metadata: dict) -> str:
```

Comment:
Do we need this new function or could you use this to replace the `check_for_matching_filetype` function above this? Since we have such a rigid structure now, I'm not sure we even need the config dictionary file.

```python
mission, instrument, data_level, descriptor, startdate, enddate = filename.split("_")
# create your path from these, maybe raise if something doesn't match our requirements
```

Reply:
`check_for_matching_filetype` checks that the filename follows one of the pattern in the `config.json`. Once it finds that format, then it stops there and continue. If we remove `check_for_matching_filetype`, then having `config.json` is not useful.

Reply:
Yeah, so up to you whether you think we'll need that check/config, but now that we are more rigid in the filename requirements I'm not so sure we will was my suggestion to merge them together. Adding onto my above suggestion we could add checks from the config file directly here as well:
```python
mission, instrument, data_level, descriptor, startdate, enddate = filename.split("_")
# create your path from these, maybe raise if something doesn't match our requirements
if not enddate.endswith((".pkts", ".cdf")):
    raise ValueError("Bad filetype, the SDC requires '.pkts' or '.cdf' filetypes to be uploaded")
if instrument not in (...):
    raise ValueError("Acceptable instruments are: ...")
# Even more validation on things if you want as well.
file_path_to_upload = same_as_you_have_below
```